crates/oxc_language_server/src/main.rs

```diff
@@ -492,21 +530,46 @@ impl Backend {
         if config.exists() {
             config_path = Some(config);
         }
+
+        let mut oxlintrc = None;
+        let mut config_store = None;
+        let mut linter = self.server_linter.write().await;
         if let Some(config_path) = config_path {
-            let mut linter = self.server_linter.write().await;
-            let config = Oxlintrc::from_file(&config_path)
-                .expect("should have initialized linter with new options");
-            let config_store = ConfigStoreBuilder::from_oxlintrc(true, config.clone())
-                .expect("failed to build config")
-                .build()
-                .expect("failed to build config");
+            oxlintrc = Some(
+                Oxlintrc::from_file(&config_path)
+                    .expect("should have initialized linter with new options"),
+            );
+            config_store = Some(
+                ConfigStoreBuilder::from_oxlintrc(true, oxlintrc.clone().unwrap())
+                    .expect("failed to build config")
+                    .build()
+                    .expect("failed to build config"),
+            );
+        }
+
+        if self.options.lock().await.use_nested_configs {
+            let mut config_files: FxHashMap<PathBuf, ConfigStore> = FxHashMap::default();
+            let existing_nested_configs = (*self.nested_configs.read().await).clone();
+            for (key, value) in existing_nested_configs {
+                config_files.insert(key.clone(), value.clone());
+            }
```

Comment:
Copying the map so that I can pass it into `Linter::new_with_nested_configs`. Might be a better way to do the copy that I'm not aware of.

Reply:
maybe std::mem::take? https://doc.rust-lang.org/std/mem/fn.take.html but not sure if that works in async context like this